packages/runtime/container-runtime/src/garbageCollection.ts

```diff
@@ -320,14 +323,15 @@ export class GarbageCollector implements IGarbageCollector {
         } else {
             // For new documents, GC has to be exlicitly enabled via the gcAllowed flag in GC options.
             this.gcEnabled = gcOptions.gcAllowed === true;
-            this.sessionExpiryTimeoutMs = this.gcOptions.gcTestSessionTimeoutMs;
+            // Set the Session Expiry only if the flag is enabled or the test option is set.
+            if (this.mc.config.getBoolean(runSessionExpiry)) {
```

Comment:
Add this.gcEnabled check here. Basically, only have session expiry if GC is enabled.